eng/pipelines/templates/steps/build-extended-artifacts.yml

```diff
@@ -0,0 +1,105 @@
+parameters:
+  - name: BeforePublishSteps
+    type: object
+    default: []
+  - name: TestPipeline
+    type: boolean
+    default: false
+  - name: ServiceDirectory
+    type: string
+    default: ''
+  - name: BuildDocs
+    type: boolean
+    default: true
+  - name: Artifacts
+    type: object
+    default: []
+  - name: DevFeedName
+    type: string
+    default: 'public/azure-sdk-for-python'
+# The variable TargetingString is set by template `eng/pipelines/templates/steps/targeting-string-resolve.yml`. This template is invoked from yml files:
+#     eng/pipelines/templates/jobs/ci.tests.yml
+#     eng/pipelines/templates/jobs/ci.yml
+#     eng/pipelines/templates/jobs/live.test.yml
+
+# Please use `$(TargetingString)` to refer to the python packages glob string. This was previously `${{ parameters.BuildTargetingString }}`.
+steps:
+  - task: DownloadPipelineArtifact@2
+    inputs:
+      artifactName: 'packages'
+      targetPath: $(Build.ArtifactStagingDirectory)
+
+  - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
+    parameters:
+      PackageName: "azure-template"
+      ServiceDirectory: "template"
+      TestPipeline: ${{ parameters.TestPipeline }}
+
+  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+
+  - script: |
+      echo "##vso[build.addbuildtag]Scheduled"
+    displayName: 'Tag scheduled builds'
+    condition: and(eq(variables['Build.SourceBranchName'], variables['DefaultBranch']), eq(variables['Build.Reason'],'Schedule'))
+
+  - task: UsePythonVersion@0
+    displayName: 'Use Python $(PythonVersion)'
+    inputs:
+      versionSpec: $(PythonVersion)
+
+  - script: |
+      python -m pip install setuptools==58.3.0
+      python -m pip install -r eng/ci_tools.txt
+    displayName: 'Prep Environment'
+
+  - template: set-dev-build.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+
+  - ${{if eq(variables['System.TeamProject'], 'internal') }}:
+    - template: auth-dev-feed.yml
+      parameters:
+        DevFeedName: ${{ parameters.DevFeedName }}
+
+  - task: PythonScript@0
+    displayName: 'Generate Docs'
+    condition: and(succeededOrFailed(), ${{parameters.BuildDocs}})
+    inputs:
+      scriptPath: 'scripts/devops_tasks/setup_execute_tests.py'
+      arguments: >-
+        "$(TargetingString)" 
+        --service="${{ parameters.ServiceDirectory }}" 
+        --toxenv=sphinx
+
+  - pwsh: |
+      Write-Host "##vso[task.setvariable variable=PIP_INDEX_URL]https://pypi.python.org/simple"
+    displayName: Reset PIP Index For APIStubGen
+
+  - template: /eng/pipelines/templates/steps/run_apistub.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+
+  - ${{ parameters.BeforePublishSteps }}
+
+  - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
+    parameters:
+      ArtifactPath: '$(Build.ArtifactStagingDirectory)'
+      ArtifactName: 'packages_extended'
```

Comment:
What do we expect to be in here? Do we have any release steps that need to adjust to pulling things from here now?

Reply:
APIView, Docs.

Reply:
Do we need to update other stages to pull from this new packages_extended artifact? In particular release stages?

Reply:
This PR has been caught in waiting so long that it's hard to remember, but I definitely tested the release process with the template PR at some point. I'll verify that before I merge.

Reply:
[Re-verification run](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2839899&view=results)

Reply:
[Re-verification run #2](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2840062&view=results)